HLTrigger/HLTcore/src/HLTPrescaleProvider.cc

```diff
@@ -102,46 +102,33 @@ int HLTPrescaleProvider::prescaleSet(const edm::Event& iEvent, const edm::EventS
   }
 }
 
-unsigned int HLTPrescaleProvider::prescaleValue(const edm::Event& iEvent,
-                                                const edm::EventSetup& iSetup,
-                                                const std::string& trigger) {
-  const int set(prescaleSet(iEvent, iSetup));
-  if (set < 0) {
-    return 1;
-  } else {
-    return hltConfigProvider_.prescaleValue(static_cast<unsigned int>(set), trigger);
-  }
+template<>
+FractionalPrescale 
+HLTPrescaleProvider::convertL1PS<FractionalPrescale>(double val)const{
+  int numer = static_cast<int>(val*kL1PrescaleDenominator+0.5);
```

Comment:
Question:
Should we check that `val*kL1PrescaleDenominator` is an integer ?
It *should* be, but if it isn't (for whatever reason) wouldn't it be better to catch it ?

Technically, "is an integer" could be just
```c++
if (numer != val*kL1PrescaleDenominator) ...
```
or allow for some small rounding, for example
```c++
static constexpr double kL1RoundingEpsilon = 0.001;
if (std::abs(numer - val*kL1PrescaleDenominator) > kL1RoundingEpsilon) ...
```

Reply:
hmm a good point. Yes we should. Exception or just set it -1?

Reply:
I'd return the approximate value ad issue a LogWarning